docs/fornax-design/mizar_gateway_design

```diff
@@ -0,0 +1,54 @@
+# Adding Cross Cluster Gateways to Mizar
+
+A gateway is a field belonging to a mizar subnet object that all the traffics go to the gateway if there are any cross cluster requests. 
+
+```
+- name: Gateway
+  type: string
+  priority: 0
+  JSONPath: .spec.gateway
+  description: The IP of the gateway
+```
+
+Also, we add another field External to the subnet object that indicates wether the subnet belongs to the local cluster
+
+```
+- name: External
+  type: boolean
+  priority: 0
+  JSONPath: .spec.external
+  description: True means that the subnet does not belong to the local cluster; False means that the subnet belongs to the local cluster
+```
+
+## Control Plane
+For mizar control planes, we add the two fields above to subnet objects. The default value of gateway is empty and the one of external is false. If all the requests happen within the clusters, the two fields do not invovle at all.
+
+### Create a gateway droplet
+1. Deploy gateway endpoints
+2. Get gateway endpoint droplet ip 
+3. Add the gateway droplet ip to subnet
+4. The change triggers subnet workflows to delete the droplet and treat it as gateway droplet
```

Comment:
this is not consistent with our POC method. what's the reason of the divergence?

Reply:
POC is to have a predefined host as gateway. I just make it configurable. After we deploy a gateway and know its ip, we will update the subnet by adding the gateway ip. That ip will be removed from droplet set as POC does